pyanaconda/modules/boss/installation.py

```diff
@@ -0,0 +1,151 @@
+#
+# Copyright (C) 2021 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+import os.path
+import shutil
+import glob
+
+from pyanaconda.anaconda_loggers import get_module_logger
+from pyanaconda.core import util
+from pyanaconda.flags import flags
+from pyanaconda.modules.common.task import Task
+
+
+log = get_module_logger(__name__)
+
+ANACONDA_LOG_DIR = "/var/log/anaconda/"
+
+
+class CopyLogsTask(Task):
+    """Task to copy logs to target system."""
+    def __init__(self, sysroot):
+        """Create a new task.
+
+        :param sysroot: a path to the root of installed system
+        :type sysroot: str
+        """
+        super().__init__()
+        self._sysroot = sysroot
+
+    @property
+    def name(self):
+        return "Copy installation logs"
+
+    def run(self):
+        self._copy_logs()
+        self._copy_kickstart()
+        self._relabel_files()
+
+    def _copy_logs(self):
+        """Copy installation logs to the target system"""
+        log.info("Copying logs from the installation environment...")
+
+        if flags.nosave_logs:
```

Comment:
You should use the `conf` object. I guess you wanted to start with a task and implement the configuration options later. That's not possible. Flags are not synchronized or shared between modules, so they are valid only in the UI process. That's why we have created the configuration files. This flag will be always set to False in a DBus module. I think you can create the configuration options in another pull request and then rebase this one.